src/Dom/Document.php

```diff
@@ -327,15 +379,23 @@ public function __construct($version = '', $encoding = null)
     /**
      * Named constructor to provide convenient way of transforming HTML into DOM.
      *
-     * @param string $html     HTML to turn into a DOM.
-     * @param string $encoding Optional. Encoding of the provided HTML string.
+     * @param string       $html    HTML to turn into a DOM.
+     * @param array|string $options Optional. Array of options to configure the document. Used as encoding if a string
+     *                              is passed. Defaults to an empty array.
      * @return Document|false DOM generated from provided HTML, or false if the transformation failed.
      */
-    public static function fromHtml($html, $encoding = null)
+    public static function fromHtml($html, $options = [])
     {
+        // Assume options are the encoding if a string is passed, for BC reasons.
+        if (is_string($options)) {
+            $options = [self::OPTION_ENCODING => $options];
+        }
+
+        $encoding = isset($options[self::OPTION_ENCODING]) ? $options[self::OPTION_ENCODING] : null;
+
         $dom = new self('', $encoding);
```

Comment:
Alternatively:

```suggestion
        $options = array_merge( self::KNOWN_OPTIONS, $options );

        $dom = new self('', $options[self::OPTION_ENCODING]);
```

Reply:
But I see this merge is already being done below.

Reply:
Yes, I wanted to avoid multiple merges.